src/header/context.rs

```diff
@@ -0,0 +1,76 @@
+use std::fmt;
+
+/// A [`Context`](../block/struct.Context.html)'s cosmetic label.
+#[derive(Clone, Copy, PartialEq, Eq, Debug)]
+pub enum ContextLabel {
+    Describe,
+    Context,
+    Specify,
+    Given,
+    When,
+}
+
+impl fmt::Display for ContextLabel {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            &ContextLabel::Describe => write!(f, "Describe"),
+            &ContextLabel::Context => write!(f, "Context"),
+            &ContextLabel::Specify => write!(f, "Specify"),
+            &ContextLabel::Given => write!(f, "Given"),
+            &ContextLabel::When => write!(f, "When"),
+        }
+    }
+}
+
+/// A [`Header`](trait.Header.html) with label and name of a [`Context`](../block/struct.Context.html).
+#[derive(Clone, PartialEq, Eq, Debug)]
+pub struct ContextHeader {
+    pub label: ContextLabel,
+    pub name: &'static str,
```

Comment:
question: shouldn't this be a `String` ? With this type bound you constrain how the user can declare suites

Reply:
It certainly would be more flexible. But then again I'm not sure I'd consider dynamically generated tests (i.e. at runtime) a good practice to begin with. Tests should be K.I.S.S., no? I wouldn't want to have to search for the piece of code that dynamically generated the test case that causes my suite to fail.

Reply:
This is not to say I'd not be fine with changing labels back to being `String`.